tests/test_sourceTable.py

```diff
@@ -662,13 +667,22 @@ def _testFluxSlot(self, slotName):
         self.assertEqual(getattr(record, "get%sErr" % (instFluxName,))(), err)
         self.assertEqual(getattr(record, "get%sFlag" % (slotName,))(), flag)
 
+        # Should also be able to retrieve them as columns from the catalog
+        self.assertEqual(getattr(catalog, f"get{instFluxName}")()[0], instFlux)
+        self.assertEqual(getattr(catalog, f"get{instFluxName}Err")()[0], err)
+
         # And we should be able to delete the slot, breaking the mapping.
         table.schema.getAliasMap().erase("slot_%s" % (slotName,))
         self.assertFalse(getattr(table, "get%sSlot" % (slotName,))().isValid())
         self.assertNotEqual(getattr(table, "get%sSlot" % (slotName,))().getMeasKey(), instFluxKey)
         self.assertNotEqual(getattr(table, "get%sSlot" % (slotName,))().getErrKey(), errKey)
         self.assertNotEqual(getattr(table, "get%sSlot" % (slotName,))().getFlagKey(), flagKey)
 
+        with self.assertRaises(lsst.pex.exceptions.LogicError):
+            getattr(catalog, f"get{instFluxName}")()
+        with self.assertRaises(lsst.pex.exceptions.LogicError):
```

Comment:
Why should these two raise? I would have thought they would work?

Reply:
These are referring to keys which are defined as part of the slot. When the slot is erased (at line 675, above), all the keys defined as part of it are invalidated. Accessing via those invalid keys raises.

I'll add a comment to that effect.